klippy/configfile.py

```diff
@@ -149,33 +149,75 @@ def _strip_duplicates(self, data, config):
                 is_dup_field = True
                 lines[lineno] = '#' + lines[lineno]
         return "\n".join(lines)
-    def _build_config_wrapper(self, data):
-        # Strip trailing comments from config
+    def _parse_config_buffer(self, buffer, filename, fileconfig):
+        if not buffer:
+            return
+        data = '\n'.join(buffer)
+        del buffer[:]
+        sbuffer = StringIO.StringIO(data)
+        fileconfig.readfp(sbuffer, filename)
+    def _resolve_include(self, source_filename, include_spec, fileconfig,
+                         visited):
+        dirname = os.path.dirname(source_filename)
+        include_spec = include_spec.strip()
+        include_glob = os.path.join(dirname, include_spec)
+        include_filenames = glob.glob(include_glob)
+        if not include_filenames and not glob.has_magic(include_glob):
+            # Empty set is OK if wildcard but not for direct file reference
+            raise error("Include file '%s' does not exist", include_glob)
+        include_filenames.sort()
+        for include_filename in include_filenames:
+            include_data = self._read_config_file(include_filename)
+            self._parse_config(include_data, include_filename, fileconfig,
+                               visited)
+        return include_filenames
+    def _parse_config(self, data, filename, fileconfig, visited = set()):
```

Comment:
Assigning `visited = set()` here effectively makes visited a global variable and that breaks the RESTART command.  A new set (or just a dictionary) should be passed on each _parse_config() call.

Reply:
Fixed this too